modal/_utils/async_utils.py

```diff
@@ -494,8 +494,12 @@ async def aclosing(
 
 async def sync_or_async_iter(iterable: Union[Iterable[T], AsyncIterable[T]]) -> AsyncGenerator[T, None]:
     if hasattr(iterable, "__aiter__"):
-        async for item in typing.cast(AsyncIterable[T], iterable):
-            yield item
+        agen = typing.cast(AsyncGenerator[T, None], iterable)
+        try:
+            async for item in agen:
+                yield item
+        finally:
+            await agen.aclose()
```

Comment:
This would crash if a non-generator iterable is sent to the function - simplest fix is probably to just make the function only accept iterable | asyncgenerator for now